### PR TITLE
Resource manager appointments calendar

### PR DESCRIPTION
**Pull request by benlovell**
Adds the basis of the resource manager's appointments calendar. From here they
can currently move appointments around in terms of timing and the guider
carrying them out.

The commits just break things up a bit as there is a whole bunch of code here.
I'll squash things down after the review.

### CODE REVIEW

File: app/controllers/appointments_controller.rb [hunk 1]
```diff
@@ -55,6 +65,10 @@ def show
 
   private
 
+  def include_links?
+    params.fetch(:include_links, 'true') == 'true'
```
Comment (benlovell) [inline app/controllers/appointments_controller.rb]: It comes from params and isn't coerced.

File: app/assets/javascripts/calendars/guiders-appointments.es6 [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+      this.$el.fullCalendar('rerenderEvents');
+
+      this.checkToShowActionPanel();
+    }
+
+    undoOneChange(evt) {
+      evt.preventDefault();
+
+      const event = this.eventChanges.pop();
```
Comment (benbarnett) [inline app/assets/javascripts/calendars/guiders-appointments.es6]: I know this was lifted from another branch but we should try and make declarations at the beginning of the function.

Also, does `const` feel right here? Seems more like a `let` maybe.

File: app/assets/javascripts/calendars/guiders-appointments.es6 [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+      this.eventChanges = [];
+      this.rerenderEvents();
+
+      this.checkToShowActionPanel();
+    }
+
+    save(evt) {
+      evt.preventDefault();
+      const $hiddenInput = $('#event-changes');
```
Comment (benbarnett) [inline app/assets/javascripts/calendars/guiders-appointments.es6]: Same here, the `const` can move up a line.

File: app/assets/stylesheets/components/_calendars.scss [hunk 4]
```diff
@@ -1,5 +1,21 @@
 $calendar-today-background: $color-white;
 
+.guiders-appointments-calendar {
+  .fc-resource-cell {
+    height: 150px;
+    white-space: nowrap;
+
+    div {
+      transform: translateY(120px) rotate(-90deg);
```
Comment (benbarnett) [inline app/assets/stylesheets/components/_calendars.scss]: I think this is going to need some looking at as I get a weird layout:

screen shot 2016-10-28 at 14 18 38

cc @Guntrisoft

File: spec/support/pages/resource_calendar.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      page.driver.evaluate_script <<-JS
+        function() {
+          var calendar = $('.js-calendar');
+          var event = calendar.fullCalendar('clientEvents', #{id})[0];
+
+          event.start.hours(#{hours}).minutes(#{minutes});
+          event.end.hours(#{hours + 1}).minutes(#{minutes});
+
+          calendar.fullCalendar('updateEvent', event);
+          calendar.data('loaded-module').handleEventChange(event, function() {});
```
Comment (benbarnett) [inline spec/support/pages/resource_calendar.rb]: 👍

File: app/views/resource_calendars/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  </ol>
+
+  <h1>Appointments</h1>
+</div>
+
+<div
+  class="guiders-appointments-calendar js-calendar"
+  data-config='{"editable": true, "eventDurationEditable": true, "slotDuration": "00:10:00"}'
+  data-module="GuidersAppointmentsCalendar"
+  data-default-date="<%= Time.zone.now %>">
```
Comment (AndrewVos) [inline app/views/resource_calendars/show.html.erb]: Tempting to extract a `calendar` helper.

File: app/views/layouts/application.html.erb [hunk 7]
```diff
@@ -24,7 +24,7 @@
       type = $el.data('module'),
       config = $el.data('config');
 
-      new PWTAP[type]($el, config);
+      $el.data('loaded-module', new PWTAP[type]($el, config));
       $el.attr('data-module-loaded', true);
```
Comment (AndrewVos) [inline app/views/layouts/application.html.erb]: Can you remove this now and add this in `layouts/application.html.erb`:
```
    $('[data-module]:not([data-loaded-module])').each(function() {
```

**Comment by AndrewVos** [comment]
LGTM, other than those two comments.

**Comment by benlovell** [comment]
@AndrewVos @benbarnett Addressed those comments.

File: app/views/layouts/application.html.erb [hunk 8]
```diff
@@ -19,13 +19,12 @@
 <% content_for :body_end do %>
   <script>
   $(function() {
-    $('[data-module]:not([data-module-loaded])').each(function() {
+    $('[data-module]:not([data-loaded-module])').each(function() {
```
Comment (benlovell) [inline app/views/layouts/application.html.erb]: Thanks @andrewvos.

**Comment by benlovell** [comment]
@benbarnett Restored that back to the prior behaviour 👍

**Comment by benbarnett** [comment]
Massive 👍

**Pull request merged**